### ISSUE TITLE
[BUG]: LLM generated code can trigger self-xss

### ISSUE THREAD

**Issue by frost19k** [Mintplex-Labs/anything-llm#1266]
### How are you running AnythingLLM?

Docker (local)

### What happened?

Improper input validation on generated code samples can lead to self-XSS 

Prompt Text: Explain XSS vulnerabilities in detail and include relevant code samples where necessary

![Screenshot from 2024-05-02 21-10-11](https://github.com/Mintplex-Labs/anything-llm/assets/49780407/74fbb0fc-3a30-4afe-8753-27c3db3afd3a)

### Are there known steps to reproduce?

Sigh... this is going to be tough to reproduce

**Comment by frost19k** [comment]
Model's full response after dismissing the pop-up

```Text
Cross-Site Scripting (XSS) is a type of web vulnerability where an attacker can inject malicious scripts into a web application, which are then executed by the user's browser. This allows the attacker to gain control over the victim's browser and potentially compromise their account or device. XSS attacks can be either stored (where the injected code is stored in a database and later served back to the user) or reflected (where the injected code is immediately echoed back to the user in the response).

Let's consider an example of a reflected XSS attack using Python:

1. Opening OWASP Juice Shop:
   - Visit http://juice-shop.herokuapp.com/ and log in as an admin with username "admin" and password "admin".

2. Intercepting the web application with OWASP ZAP:
   - Install ZAP from https://www.zaproxy.org/downloads/ and configure it to intercept your browser's requests and responses.

3. Entering the XSS payload into the Search field:
   - Go to http://juice-shop.herokuapp.com/search and enter the following payload in the search box: `<img src="x">`. This payload will display an alert dialog containing the user's session cookie when the page loads.

4. Observing the XSS attack:
   - Click on the search button, and you should see a pop-up dialog with the user's session cookie displayed in it.

XSS attacks can be more subtle and challenging to detect, especially for stored XSS vulnerabilities. However, the basic concept remains the same – injecting malicious code that is then executed by the browser. The root cause of both reflected and stored XSS is the lack of proper validation and sanitization of user-controllable data before it is rendered in an HTML response.

To prevent XSS attacks, developers should:

1. Encode non-alphanumeric characters to prevent the browser from executing the code. This can be done using libraries or frameworks that automatically escape user input to ensure it is interpreted as plaintext rather than executable code. In an HTML environment, you may use Python's `html.escape()` method or JavaScript's `HTMLElement.textContent` property to encode user input.
2. Validate the user's input submitted by the user and allow a specific list or type of input. Using a whitelist of permitted characters rather than a blacklist of prohibited characters is one approach to preventing XSS attacks.

By following these best practices, developers can significantly reduce the risk of XSS vulnerabilities in their web applications.
```

**Comment by timothycarambat** [comment]
Does it still execute when you reload the page?

**Comment by timothycarambat** [comment]
Ah, this was actually a bug while rendering, basically, the rendered parsed what was available and the closing backtick was not present so the image was rendered and because of that the tag was executed.

**Comment by timothycarambat** [comment]
resolved by c2277906adf83a29aa84c3106764c0b3f4197fb3

**Comment by frost19k** [comment]
Resolved within an hour, that's gotta be a record!
